examples/fpga/artya7/util/vivado_hook_route_design_post.tcl

```diff
@@ -0,0 +1,34 @@
+open_project ../../lowrisc_ibex_top_artya7_0.1.xpr
+set saif_name "detailed_power.saif"
+
+# TODO: All cells are removed in optimization process. Needs investigation, removal happens because
+# of unloaded cells. Until it is solved, this workaround generates another implementation run where
+# optimization is skipped. That way functional post-implementation simulation can be done. create_run
+# command requires flow version, which needs to be extracted.
+regexp -- {Vivado v([0-9]{4})\.[0-9]} [version] -> year
+create_run impl_2 -parent_run synth_1 -flow "Vivado Implementation $year"
+set_property STEPS.OPT_DESIGN.IS_ENABLED false [get_runs impl_2]
+launch_runs impl_2
+wait_on_run impl_2
+open_run impl_2
+
+# Runs a post implementation functional simulation with the memory initialized with SRAMInitFile.
+# Feeds clock (100mhz) and reset switch and records switching activity for 3ms.
+if {![file exists "$saif_name"]} {
+set_property top top_artya7 [current_fileset sim_1]
+launch_simulation -mode post-implementation -type functional
+open_saif "$saif_name"
+log_saif [get_objects -r *]
+add_force {/top_artya7/IO_CLK} -radix bin {1 0ns} {0 5ns} -repeat_every 10ns
+add_force {/top_artya7/IO_RST_N} -radix bin {1 0ns}
+run 3ms
+close_saif
+}
+
+# Reporting power using .saif generated above
+open_run impl_2
+set_operating_conditions -process maximum
+read_saif "../../lowrisc_ibex_top_artya7_0.1.sim/sim_1/impl/func/xsim/$saif_name"  -out_file saif_detailed.log
+read_saif "../../lowrisc_ibex_top_artya7_0.1.sim/sim_1/impl/func/xsim/$saif_name"  -strip_path top_artya7 -out_file saif_detailed1.log
```

Comment:
This looks a little odd to me. Do we need both of these log files? If so, maybe the names could be more informative?

Reply:
It was for debugging why read_saif was not reading logged `.saif` file properly and it left there. I will remove it, thanks for noticing! Two read_saif commands was for stripping the hierarchy so that almost all the nets could be matched.